boot/zephyr/flash_map_legacy.c

```diff
@@ -101,25 +58,27 @@ int flash_area_to_sectors(int idx, int *cnt, struct flash_area *ret)
  */
 int flash_area_get_sectors(int idx, uint32_t *cnt, struct flash_sector *ret)
 {
-    uint32_t off;
-    uint32_t len;
+    const struct flash_area *fa;
     uint32_t max_cnt = *cnt;
     uint32_t rem_len;
 
-    if (flash_area_get_bounds(idx, &off, &len)) {
+    if (flash_area_open(idx, &fa))  {
```

Comment:
Please be mindful to close anything that gets opened, including in the error paths.